### PR TITLE
Support import gzip json file

### PR DESCRIPTION
**Pull request by ndinh215**
Please review my implementation of supporting gzip file.

### CODE REVIEW

**Comment by saimaz** [comment]
First of rebase all merge type commits. Let's try to keep a nice git history.

**Comment by saimaz** [comment]
You forgot to add close comment for the specific issue.

File: Command/IndexCreateCommand.php [hunk 1]
```diff
@@ -38,7 +38,13 @@ protected function configure()
                 InputOption::VALUE_NONE,
                 'If the time suffix is used, its nice to create an alias to the configured index name.'
             )
-            ->addOption('no-mapping', null, InputOption::VALUE_NONE, 'Do not include mapping');
+            ->addOption('no-mapping', 'nm', InputOption::VALUE_NONE, 'Do not include mapping')
```
Comment (saimaz) [inline Command/IndexCreateCommand.php]: Options shortcut should have not more than one letter. In this particular case let's leave this option without shortcut.

**Comment by ndinh215** [comment]
Sorry @saimaz , I will check again coveralls and add 'close issue'.

File: Command/IndexCreateCommand.php (hunk 1 above)
Comment (saimaz) [inline Command/IndexCreateCommand.php]: Oh, sorry for the comment, this commit is not related to this PR. Isn't it. There is something messed up with your git history...

**Comment by saimaz** [comment]
Please do a rebase. Now it's somehow combined with other PR's.

**Comment by ndinh215** [comment]
Got it, @saimaz

**Comment by ndinh215** [comment]
I will check again all, and inform to you on finish. Thanks.

**Comment by saimaz** [comment]
No idea what you are doing here, but here's the instruction https://akrabat.com/the-beginners-guide-to-rebasing-your-pr/

**Comment by ndinh215** [comment]
Yup, I did it. I included 2 implementations of the issue #423 & #382. 
@mvar is checking again the issue: 
- No need to create index after calling  ```$manager = $this->getManager(); ``` in ```CreateIndexCommandTest``` class. 
- Now, I still add the code check this: 
```
if (!$manager->indexExists()) {
      $manager->createIndex();
 }
```

**Comment by saimaz** [comment]
Please open a new well-formed PR with appropriate changes.

**Comment by ndinh215** [comment]
Yup, I will do it. Thanks @saimaz .
